cmd/genesis-generator/genesis-generator.go

```diff
@@ -0,0 +1,152 @@
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/offchainlabs/nitro/arbos/arbosState"
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/cmd/util/confighelpers"
+	"github.com/offchainlabs/nitro/execution/gethexec"
+	"github.com/offchainlabs/nitro/statetransfer"
+)
+
+func main() {
+	if err := mainImpl(); err != nil {
+		log.Error("Error running genesis generator", "error", err)
+		os.Exit(1)
+	}
+}
+
+func mainImpl() error {
+	args := os.Args[1:]
+	f := flag.NewFlagSet("", flag.ContinueOnError)
+
+	ConfigAddOptions(f)
+
+	k, err := confighelpers.BeginCommonParse(f, args)
+	if err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	err = confighelpers.ApplyOverrides(f, k)
+	if err != nil {
+		return fmt.Errorf("failed to apply overrides: %w", err)
+	}
+
+	var config Config
+	if err := confighelpers.EndCommonParse(k, &config); err != nil {
+		return fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	if config.GenesisJsonFile == "" {
+		return fmt.Errorf("genesis JSON file must be specified")
+	}
+	genesisJson, err := os.ReadFile(config.GenesisJsonFile)
+	if err != nil {
+		return fmt.Errorf("failed to read genesis JSON file %s: %w", config.GenesisJsonFile, err)
+	}
+	var gen core.Genesis
+	if err := json.Unmarshal(genesisJson, &gen); err != nil {
+		return fmt.Errorf("failed to unmarshal genesis JSON: %w", err)
+	}
+	var accounts []statetransfer.AccountInitializationInfo
+	for address, account := range gen.Alloc {
+		accounts = append(accounts, statetransfer.AccountInitializationInfo{
+			Addr:       address,
+			EthBalance: account.Balance,
+			Nonce:      account.Nonce,
+			ContractInfo: &statetransfer.AccountInitContractInfo{
+				Code:            account.Code,
+				ContractStorage: account.Storage,
+			},
+		})
+	}
+	initDataReader := statetransfer.NewMemoryInitDataReader(&statetransfer.ArbosInitializationInfo{
+		Accounts: accounts,
+	})
+	chainConfig := gen.Config
+	genesisArbOSInit := gen.ArbOSInit
+	serializedChainConfig, err := json.Marshal(chainConfig)
+	if err != nil {
+		return fmt.Errorf("failed to serialize chain config: %w", err)
+	}
+	parsedInitMessage := &arbostypes.ParsedInitMessage{
+		ChainId:               chainConfig.ChainID,
+		InitialL1BaseFee:      big.NewInt(config.InitialL1BaseFee),
+		ChainConfig:           chainConfig,
+		SerializedChainConfig: serializedChainConfig,
+	}
+	genesisHash, err := generateGenesisHash(rawdb.NewMemoryDatabase(),
+		gethexec.DefaultCacheConfigFor(&config.Caching),
+		initDataReader,
+		chainConfig,
+		genesisArbOSInit,
+		parsedInitMessage,
+		config.AccountsPerSync,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to generate genesis hash: %w", err)
+	}
+	fmt.Printf("Genesis block hash: %s\n", genesisHash.Hex())
```

Comment:
let's print the full assertion:
blockhash, sendroot (can be taken from block header, but will be 0), Batch # (will be 1) pos in batch (will be 0)